src/status_im/ui/screens/wallet/components/views.cljs

```diff
@@ -52,27 +54,80 @@
           (when-not (nil? error)
             [tooltip error])]]))))
 
-(views/defview view-currency [style]
-  (views/letsubs [visible-tokens [:wallet.settings/visible-tokens]
-                  symbol         [:wallet.send/symbol]]
-    [react/view
-     [react/text {:style styles/label} (i18n/label :t/currency)]
-     [react/view (merge styles/currency-container
-                        style)
-      [react/text
-       (name symbol)]]]))
+(views/defview view-asset [ symbol]
+  [react/view
+   [react/text {:style styles/label}
+    (i18n/label :t/wallet-asset)]
+   [react/view styles/asset-container
+    [react/text
+     (name symbol)]]])
 
-(views/defview choose-currency [{:keys [style on-change value]}]
-  (views/letsubs [visible-tokens [:wallet.settings/visible-tokens]]
-    [react/view
-     [react/text {:style styles/label} (i18n/label :t/currency)]
-     [react/view (merge styles/currency-container
-                        style)
-      [react/picker {:selected   value
-                     :style      {:color "white"}
-                     :item-style styles/wallet-name
-                     :on-change  on-change}
-       (map (fn [s] {:value (name s) :color "white"}) (conj visible-tokens (:symbol tokens/ethereum)))]]]))
+(defn- type->handler [k]
+  (case k
+    :send    :wallet.send/set-symbol
+    :request :wallet.request/set-symbol
+    (throw (str "Unknown type: " k))))
+
+(defn- render-token [{:keys [symbol name icon decimals]} balance type]
+  [list/touchable-item  #(do (re-frame/dispatch [(type->handler type) symbol])
+                             (re-frame/dispatch [:navigate-back]))
```

Comment:
I don't like this double dispatch very much - instead of double dispatch + special preload logic for `:navigate-back` (hard to track, on 2 places), we should create one atomic handler for that kind of preset/navigation.

Reply:
Make a todo and an tech-debt issue that links to it @jeluard let's not delay this merge too much

Reply:
I agree but didn't want to make too big changes. This mimic what is done for the `send` screen which share most of the logic.